newton/_src/solvers/semi_implicit/kernels_particle.py

```
@@ -0,0 +1,508 @@
+    rest = spring_rest_lengths[tid]
+
+    xi = x[i]
+    xj = x[j]
+
+    vi = v[i]
+    vj = v[j]
+
+    xij = xi - xj
+    vij = vi - vj
+
+    l = wp.length(xij)
```

Ruff (0.13.3)

Ambiguous variable name: `l`

(E741)